File: Team5/ReverseOnlyLetters.cpp
class Solution {
public:
    string reverseOnlyLetters(string S) {
      int start=0;
      int end=S.size();

      while(start<end){
        if(isalpha(S[start]) && isalpha(S[end]){
            char m=S[start];
            S[start]=S[end];
            S[end]=m;
            start++;
            end--;
        }
        else if(isalpha(S[start]) && !isalpha(S[end])){
                end--;
        }
        else if(!isalpha(S[start])){
             start++;
        }
     }
      return S;
    }
};
           

File: Team1/RiyaSingh/reverseOnlyLetters.java
class Solution {
    public String reverseOnlyLetters(String S) {
        int left = 0;
        int right = S.length() - 1;
        char[] str = S.toCharArray();
        while(left<=right){
            if(!Character.isLetter(S.charAt(left))){
                left++;
                continue;
            }else if(!Character.isLetter(S.charAt(right))){
                right--;
                continue;
            }
            char sub = str[left];
            str[left] = str[right];
            str[right] = sub;
            left++;
            right--;
        }
        String string = String.copyValueOf(str);
        return string;
    }
}
